contracts/plugins/assets/origin/OETHCollateralL2Base.sol

```diff
@@ -0,0 +1,190 @@
+
+    function decimals() external view returns (uint8);
+
+    function decreaseAllowance(address spender, uint256 subtractedValue) external returns (bool);
+
+    function deposit(uint256 assets, address receiver) external returns (uint256);
+
+    function governor() external view returns (address);
+
+    function increaseAllowance(address spender, uint256 addedValue) external returns (bool);
+
+    function initialize() external;
+
+    function isGovernor() external view returns (bool);
+
+    function maxDeposit(address) external view returns (uint256);
+
+    function maxMint(address) external view returns (uint256);
+
+    function maxRedeem(address owner) external view returns (uint256);
+
+    function maxWithdraw(address owner) external view returns (uint256);
+
+    function mint(uint256 shares, address receiver) external returns (uint256);
+
+    function name() external view returns (string memory);
+
+    function previewDeposit(uint256 assets) external view returns (uint256);
+
+    function previewMint(uint256 shares) external view returns (uint256);
+
+    function previewRedeem(uint256 shares) external view returns (uint256);
+
+    function previewWithdraw(uint256 assets) external view returns (uint256);
+
+    function redeem(
+        uint256 shares,
+        address receiver,
+        address owner
+    ) external returns (uint256);
+
+    function symbol() external view returns (string memory);
+
+    function totalAssets() external view returns (uint256);
+
+    function totalSupply() external view returns (uint256);
+
+    function transfer(address recipient, uint256 amount) external returns (bool);
+
+    function transferFrom(
+        address sender,
+        address recipient,
+        uint256 amount
+    ) external returns (bool);
+
+    function transferGovernance(address _newGovernor) external;
+
+    function transferToken(address asset_, uint256 amount_) external;
+
+    function withdraw(
+        uint256 assets,
+        address receiver,
+        address owner
+    ) external returns (uint256);
+}
+
+interface IMorphoChainlinkOracleV2 {
+    function price() external view returns (uint256);
+}
+
+/**
+ * @title Origin Staked ETH Collateral for Base L2
+ * @notice Collateral plugin for Origin OETH,
+ * tok = wsuperOETHb  (wrapped superOETHb)
+ * ref = superOETHb (pegged to ETH 1:1)
+ * tar = ETH
+ * UoA = USD
+ */
+contract OETHCollateralL2Base is AppreciatingFiatCollateral {
+    using OracleLib for AggregatorV3Interface;
+    using FixLib for uint192;
+
+    IMorphoChainlinkOracleV2 public immutable targetPerTokChainlinkFeed; // {tar/token}
+
+    AggregatorV3Interface public immutable uoaPerTargetChainlinkFeed; // {UoA/tar}
+    uint48 public immutable uoaPerTargetChainlinkTimeout; // {s}
+
+    /// @param config.chainlinkFeed - ignored
+    /// @param config.oracleTimeout - ignored
+    /// @param config.oracleError {1} Should be the oracle error for UoA/tok
```

Comment:
this should be the combined oracle error I think, since it is applied post-product